core/src/processing/opengl/PSurfaceJOGL.java

```diff
@@ -273,6 +273,15 @@ protected void initGL() {
       } else if (PJOGL.profile == 2) {
         try {
           profile = GLProfile.getGL2ES2();
+
+          // workaround for https://jogamp.org/bugzilla/show_bug.cgi?id=1347
+          if (!profile.isHardwareRasterizer()) {
+            GLProfile hardware = GLProfile.getMaxProgrammable(true);
```

Comment:
please double check that GLProfile hardware = GLProfile.getMaxProgrammable(true); gives you GLES2 since I think you may en up with the Mesa 3D software renderer because Mesa 3D do support a very high Programmable profile using software renderer.